### PR TITLE
Added kueuectl-docs generator.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Added kueuectl-docs generator.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Added documentation for kubectl-kueue plugin.
```

### CODE REVIEW

**Comment by mbobrovskyi** [comment]
make /test

File: cmd/kueuectl-docs/generators/doc.go [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+	cmd.InitDefaultHelpFlag()
+	name := cmd.CommandPath()
+	short := cmd.Short
+	long := cmd.Long
+
+	if len(long) == 0 {
+		long = short
+	}
+
+	if _, err := fmt.Fprintf(w, "---\ntitle: %s\ncontent_type: tool-reference\nweight: 30\nauto_generated: true\n", name); err != nil {
```
Comment (trasc) [inline cmd/kueuectl-docs/generators/doc.go:114]: why `weight: 30` ? :)

File: cmd/kueuectl-docs/generators/doc.go [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+	}
+	return false
+}
+
+func flagUsages(f *pflag.FlagSet) string {
+	x := new(bytes.Buffer)
+
+	lines := make([]string, 0)
+
+	lines = append(lines, "   <table style=\"width: 100%; table-layout: fixed;\">\n<colgroup>\n"+
```
Comment (trasc) [inline cmd/kueuectl-docs/generators/doc.go:262]:
```suggestion
	lines = append(lines, "<table style=\"width: 100%; table-layout: fixed;\">\n<colgroup>\n"+
```

File: cmd/kueuectl-docs/generators/doc.go [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+		name = "&lt;a string in the form 'file:N'&gt;"
+	}
+	return
+}
+
+func processUsage(usage string) string {
+	var buf bytes.Buffer
+	var result string
+	usage = strings.ReplaceAll(usage, "<", "&lt;")
+	usage = strings.ReplaceAll(usage, ">", "&gt;")
```
Comment (trasc) [inline cmd/kueuectl-docs/generators/doc.go:426]: html.EscapeString

**Review by trasc — COMMENTED** [review]
I think using https://pkg.go.dev/html/template cam make it a lot easier to follow  and maintain the implementation.

File: cmd/kueuectl-docs/generators/doc.go [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+	// Print the "generated" warning
+	fmt.Fprintf(w, "%s\n\n", generatedWarning)
+
+	if _, err := fmt.Fprintf(w, "%s\n\n", `## Synopsis`); err != nil {
+		return err
+	}
+
+	// Escape any '<', '>' characters found in the long description
+	long = strings.ReplaceAll(long, "<", "&lt;")
+	long = strings.ReplaceAll(long, ">", "&gt;")
```
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply cmd/kueuectl-docs/generators/doc.go]: Done

File: cmd/kueuectl-docs/generators/doc.go (hunk 1 above)
Comment (mbobrovskyi) [reply cmd/kueuectl-docs/generators/doc.go]: Removed

File: cmd/kueuectl-docs/generators/doc.go (hunk 2 above)
Comment (mbobrovskyi) [reply cmd/kueuectl-docs/generators/doc.go]: Done

File: hack/internal/tools/kueuectl-docs/generators/doc.go [hunk 5]
```diff
@@ -0,0 +1,374 @@
...
+		return "uint"
+	case "stringSlice", "stringArray":
+		return "strings"
+	case "intSlice":
+		return "ints"
+	case "uintSlice":
+		return "uints"
+	case "boolSlice":
+		return "bools"
+	case "mapStringString":
```
Comment (trasc) [inline hack/internal/tools/kueuectl-docs/generators/doc.go:223]: should this be `stringToString`?

File: hack/internal/tools/kueuectl-docs/generators/doc.go [hunk 6]
```diff
@@ -0,0 +1,374 @@
...
+		return "&lt;comma-separated 'key=value' pairs&gt;"
```
Comment (trasc) [inline hack/internal/tools/kueuectl-docs/generators/doc.go:224]: Do we still need to escape this if we are using templates?

Comment (mbobrovskyi) [reply hack/internal/tools/kueuectl-docs/generators/doc.go]: Thanks

File: hack/internal/tools/kueuectl-docs/generators/doc.go (hunk 5 above)
Comment (mbobrovskyi) [reply hack/internal/tools/kueuectl-docs/generators/doc.go]: You right. Fixed. Also fixed another cases.

**Review by trasc — COMMENTED** [review]
/lgtm

**Comment by alculquicondor** [comment]
What is the context here?
What is this based on and where would it output the documentation?

What is the benefit of autogenerating, as opposed to keeping a minimal documentation and directing users to use `--help` to learn more?

**Comment by mbobrovskyi** [comment]
IIRC we discussed to use minimal documentation, because it's very hard to add all details manually.

I think it can be useful. 
 - The documentation will be consistent, so we won't miss something on follow ups.
 - User doesn't need to install CLI to check full documentation.
 - We can also use generator code to generate documentation for kjobctl too.

File: Makefile [hunk 7]
```diff
@@ -320,8 +320,21 @@ importer-image: importer-image-build
 
 .PHONY: kueuectl
 kueuectl:
-	$(GO_BUILD_ENV) $(GO_CMD) build -ldflags="$(LD_FLAGS)" -o bin/kubectl-kueue cmd/kueuectl/main.go
+	$(GO_BUILD_ENV) $(GO_CMD) build -ldflags="$(LD_FLAGS)" -o $(PROJECT_DIR)/bin/kubectl-kueue cmd/kueuectl/main.go
+
+.PHONY: kueuectl-docs
```
Comment (alculquicondor) [inline Makefile:325]: This directive should be in Makefile-deps.mk, not here

Comment (mbobrovskyi) [reply Makefile]: Done

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/hold cancel

**Comment by mbobrovskyi** [comment]
/release-note-edit

```release-note
NONE
```

Because it was reverted in in #2611.

**Pull request merged**